source/extensions/filters/http/oauth2/oauth_client.h

```diff
@@ -69,6 +70,7 @@ class OAuth2ClientImpl : public OAuth2Client, Logger::Loggable<Logger::Id::upstr
 
   Upstream::ClusterManager& cm_;
   const envoy::config::core::v3::HttpUri uri_;
+  const absl::optional<std::chrono::seconds> default_expires_in_;
```

Comment:
```suggestion
  const std::chrono::seconds> default_expires_in_;
```

I wasn't sure whether to use `optional` here. Is it better to use zero as an implicit null value? Since setting `absl::optional<std::chrono::seconds>(0s)` doesn't make much sense.

Reply:
IMO I would just use a default 0/null value that we check against in the token examination step (in `onSuccess`, like you basically already do), and skip `optional` in favor of always setting and using the default proto value if absent. @rgs1 might have more opinions

Reply:
Sounds good, done!